lib/shared/src/models/sync.ts

```diff
@@ -532,24 +522,22 @@ async function fetchServerSideModels(
 }
 
 /**
- * maybeAdjustContextWindows adjusts the context window input tokens for specific models to prevent
- * context window overflow caused by token count discrepancies.
- *
- * Currently, the OpenAI tokenizer is used by default for all models. However, it often
- * counts tokens incorrectly for non-OpenAI models (e.g., Mistral), leading to over-counting
- * and potentially causing completion requests to fail due to exceeding the context window.
+ * Adjusts context windows for models based on user tier and model characteristics.
  *
- * The proper fix would be to use model-specific tokenizers, but this would require significant
- * refactoring. As a temporary workaround, this function reduces the `maxInputTokens` for specific
- * models to mitigate the risk of context window overflow.
+ * This function:
+ * 1. Applies tokenizer-specific adjustments (e.g., for Mistral models)
+ * 2. Enforces tier-specific context window limits (free, pro, enterprise)
+ * 3. Sets appropriate output token limits based on model capabilities
  *
- * @param {ServerModel[]} models - An array of models from the site config.
- * @returns {ServerModel[]} - The array of models with adjusted context windows where applicable.
+ * @param models - Array of models to adjust
+ * @param isPro - Whether the user has a pro subscription
+ * @param isFree - Whether the user has a free subscription
+ * @returns Array of models with adjusted context windows
  */
 export const maybeAdjustContextWindows = (
     models: ServerModel[],
     isPro = false,
-    isEnterprise = false,
+    isFree = false,
```

Comment:
```suggestion
    config: { tier: 'free' | 'pro' | 'enterprise'; longContextWindowFlagEnabled: boolean }
```

I think this is hard to follow and understand if we have 3 different params that are boolean. We should just update it to use an object instead